heat/core/indexing.py

```diff
@@ -56,14 +56,14 @@ def nonzero(x: DNDarray) -> DNDarray:
     except AttributeError:
         raise TypeError("Input must be a DNDarray, is {}".format(type(x)))
 
+    lcl_nonzero = torch.transpose(torch.nonzero(input=x.larray, as_tuple=False), 0, 1)
+
     if x.split is None:
         # if there is no split then just return the values from torch
-        lcl_nonzero = torch.nonzero(input=local_x, as_tuple=False)
         gout = list(lcl_nonzero.size())
         is_split = None
     else:
         # a is split
-        lcl_nonzero = torch.nonzero(input=local_x, as_tuple=False)
         # adjust local indices along split dimension
         _, displs = x.counts_displs()
         lcl_nonzero[..., x.split] += displs[x.comm.rank]
```

Comment:
ll. 68-69 are based on the assumption that the global nonzero is distributed along the rows, but now it's been transposed, so these need to be updated - or the transposition needs to happen afterwards